src/main/java/org/spongepowered/common/mixin/core/entity/merchant/villager/AbstractVillagerEntityMixin.java

```diff
@@ -0,0 +1,92 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.mixin.core.entity.merchant.villager;
+
+import net.minecraft.entity.merchant.villager.AbstractVillagerEntity;
+import net.minecraft.entity.merchant.villager.VillagerEntity;
+import net.minecraft.entity.merchant.villager.VillagerTrades;
+import net.minecraft.item.MerchantOffer;
+import net.minecraft.item.MerchantOffers;
+import org.spongepowered.api.data.type.Profession;
+import org.spongepowered.api.entity.living.trader.Villager;
+import org.spongepowered.api.item.merchant.TradeOffer;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.Redirect;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.common.mixin.core.entity.AgeableEntityMixin;
+import org.spongepowered.common.registry.SpongeVillagerRegistry;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Mixin(AbstractVillagerEntity.class)
+public abstract class AbstractVillagerEntityMixin extends AgeableEntityMixin {
+    private List<MerchantOffer> impl$newOffers = new ArrayList<>();
+
+    /**
+     * @author i509VCB - February 17th, 2020 - 1.14.4
+     *
+     * @reason Here we apply the Trade Mutators. For a villager we will apply our mutators and then add them to the villager's merchant offers.
+     * If a wandering trader is the subject here, we do nothing since that merchant would already have it's new merchant offers.
+     *
+     * @todo: Do we want to support Trade Mutators on Wandering Trader's TradeOffers?
+     *
+     * @param givenMerchantOffers The merchant's current merchant offers.
+     * @param newTrades The trade factories representing the new trades this villager will receive.
+     * @param maxNumbers The maximum amount of trades for this villager to receive.
+     * @param ci CallbackInfo.
+     */
+    @Inject(method = "addTrades(Lnet/minecraft/item/MerchantOffers;[Lnet/minecraft/entity/merchant/villager/VillagerTrades$ITrade;I)V", at = @At("TAIL"))
+    private void impl$addAndApplyTradeMutators(MerchantOffers givenMerchantOffers, VillagerTrades.ITrade[] newTrades, int maxNumbers, CallbackInfo ci) {
+        if ((Object) this instanceof VillagerEntity) {
```

Comment:
So there's a bit of a elegant hack to doing this:
You can redirect to do the same as the original method, and then override this injection method in `VillagerEntityMixin` to do all this logic that otherwise you'd be doing.

I do this with WorldMixin_Tracker and ServerWorldMixin_Tracker to do extra logic only on server worlds for injections that are in the subclass.

Reply:
So something similar to how you did `impl$wrapTileEntityTick` in WorldMixin_Tracker and then override it in ServerWorldMixin_Tracker

Reply:
Yep, precisely